system/CodeIgniter.php

```diff
@@ -528,7 +528,7 @@ protected function detectEnvironment()
     {
         // Make sure ENVIRONMENT isn't already set by other means.
         if (! defined('ENVIRONMENT')) {
-            define('ENVIRONMENT', $_SERVER['CI_ENVIRONMENT'] ?? 'production');
+            define('ENVIRONMENT', getenv('CI_ENVIRONMENT') ?? 'production');
```

Comment:
Why not use `env()` instead, to cover all uses?

Reply:
I like that, it essentially covers all the scenarios:
```php

    function env(string $key, $default = null)
    {
        $value = $_ENV[$key] ?? $_SERVER[$key] ?? getenv($key);

        // Not found? Return the default value
        if ($value === false) {
            return $default;
        }
...
```
And it will always be available by now:
1. Entry point (public/index.php or spark)
2. bootstrap.php
3. DotEnv.ohp
4. Common.php
5. CodeIgniter.php